pageserver/src/tenant/timeline.rs

```diff
@@ -832,6 +834,86 @@ impl Timeline {
     pub fn subscribe_for_state_updates(&self) -> watch::Receiver<TimelineState> {
         self.state.subscribe()
     }
+
+    pub fn layer_map_info(&self) -> LayerMapInfo {
+        let mut in_memory_layers = Vec::new();
+        let layer_map = self.layers.read().unwrap();
+        if let Some(open_layer) = &layer_map.open_layer {
+            in_memory_layers.push(open_layer.info());
+        }
+        for frozen_layer in &layer_map.frozen_layers {
+            in_memory_layers.push(frozen_layer.info());
+        }
+
+        let mut historic_layers = Vec::new();
+        for historic_layer in layer_map.iter_historic_layers() {
+            historic_layers.push(historic_layer.info());
+        }
+
+        LayerMapInfo {
+            in_memory_layers,
+            historic_layers,
+        }
+    }
+
+    pub async fn download_layer(&self, layer_file_name: &str) -> anyhow::Result<Option<bool>> {
+        let Some(layer) = self.find_layer(layer_file_name) else { return Ok(None) };
+        if self.remote_client.is_none() {
+            return Ok(Some(false));
+        }
+
+        let Some(remote_layer) = layer.downcast_remote_layer() else {return  Ok(Some(false)) };
```

Comment:
```suggestion
        let Some(remote_layer) = layer.downcast_remote_layer() else { return Ok(Some(false)) };
```

Interesting `rustfmt` doesn't handle this.

Reply:
Yes, `rustfmt` does not handle anything related to the feature, and `clippy` already has the lints to enforce this, odd 🙂